### DISCUSSION TITLE
How to make model "User", share a model called "Friend' with another user.

### DISCUSSION THREAD

**Discussion by dxgi** [prisma/prisma#22466] [Q&A]
How would I make it so two "User" shares one "Friend", something similar to the example above. 

An example scenario would be "User1" and "User2" are friends, basically sharing the same model called "Friend", while being accessible to user1.friends and user2.friends

```
model User {
   Friend friends[]
}

model Friend {
   User user1
   
   User user2
}
```

![image](https://github.com/prisma/prisma/assets/61899298/56c4ac5c-8ba7-4ce1-bdfe-86406b15ee97)

**Comment by ludralph** [comment]
Hi @dxgi 👋 

Thank you for raising this question.

Could you please clarify whether you are looking to create a direct many-to-many relationship between `User` model and `Friend` model, or are you seeking to establish a self-relation where users can be friends with each other without a separate `Friend` model? Additionally, do you require the ability to store additional attributes about the friendship, such as the date the friendship started, within the relationship?

**Reply by dxgi** [reply]
![image](https://github.com/prisma/prisma/assets/61899298/083920ad-ec2b-48f2-9edd-0a6dfe67f55e)

I'd also like to be able to run something similar to this
```ts
const user = await prisma.user.findFirst({
    where: {
        username: "Josh"
    },
    include: {
        friends: {
            where: {
                OR: [
                    {
                        receiver: {
                            username: 'Fred'
                        }
                    },
                    {
                        sender: {
                            username: 'Fred'
                        }
                    }
                ]
            },
            select: {
                sender: {
                    select: {
                        username: true
                    }
                },
                receiver: {
                    select: {
                        username: true
                    }
                }
            }
        }
    }
});
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
